src/uu/head/src/parse.rs

```diff
@@ -151,6 +151,7 @@ mod tests {
         }
     }
 
+    #[allow(clippy::inefficient_to_string)]
     fn obsolete_result(src: &[&str]) -> Option<Result<Vec<String>, ParseError>> {
         Some(Ok(src.iter().map(|s| s.to_string()).collect()))
```

Comment:
```suggestion
fn obsolete_result(src: &[&str]) -> Option<Result<Vec<String>, ParseError>> {
Some(Ok(src.iter().map(|&s| s.to_string()).collect()))
```

Reply:
Better to actually fix the lint

Reply:
I saw this is resolved in the main branch with a similar approach, but I prefer yours, honestly